app/lib/src/lib.rs

```diff
@@ -8,12 +8,183 @@ extern crate alloc;
 #[cfg(not(feature = "wii"))]
 pub use std::{print, println};
 
+use hecs::{World, PreparedQuery};
+use num::clamp;
+use rand::rngs::SmallRng;
+use rand::RngCore;
+use rand::{Rng, SeedableRng};
+
 #[allow(dead_code)]
 fn foo() -> bool {
     println!("Hello, world!");
     true
 }
 
+/// Represents the state of the game
+///
+/// This includes all entities, as well as potentially score, 'current mode' and other global data
+pub struct GameState {
+    pub world: World,
+}
+
+/// Represents any changes made by the outside world during a single frame.
+///
+/// Its fields are public so they can be filled in from outside the library
+/// (before `GameState::update` is called)
+#[derive(Debug, Default)]
+pub struct Changes {
+    pub controls: Controls,
+    pub delta_time_ms: u64,
+}
+
+/// Represents the state of one or multiple Wii controllers w.r.t. the game
+///
+/// Its fields are public so they can be filled in from outside the library
+/// (before `GameState::update` is called)
+#[derive(Debug, Default)]
+pub struct Controls {
+    pub home_button_down: bool,
+    pub one_button_down: bool,
+}
+
+impl GameState {
+    /// Creates a GameState in the situation at the start of the game
+    pub fn new() -> GameState {
+        let mut res = GameState { world: World::new() };
+
+        batch_spawn_entities(&mut res.world, 5);
+
+        res
+    }
+
+    /// Performs game logic of one iteration of the game loop.
+    /// Before calling this, the caller should have filled `Changes` with the required values.
+    ///
+    /// If the returned bool is `true`, the game loop should continue.
+    /// When `false` is returned, the game should quit.
+    pub fn update(&mut self, changes: &Changes) -> bool {
+        if changes.controls.home_button_down {
+            return false;
+        }
+
+        if changes.controls.one_button_down {
+            system_shake_wii(&mut self.world)
+        }
+
+        system_bounce_bounds(&mut self.world);
+        system_integrate_motion(&mut self.world);
+
+        true
+    }
+}
+
+
+
+
```

Comment:
Is clippy still working? I'd expect these whitelines to be caught by it :thinking:

Reply:
Strange. Seems like there was a race condition between the formatter and my editor.

Resolved by running the formatter manually.